### PR TITLE
CSR and CSC copy

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented copy method and copy initializer of csr and csc matrix.

### CODE REVIEW

File: tests/cupy_tests/sparse_tests/test_csr.py [hunk 1]
```diff
@@ -58,6 +58,28 @@ def test_indptr(self):
         testing.assert_array_equal(
             self.m.indptr, cupy.array([0, 2, 3, 4], self.dtype))
 
+    def test_init_copy(self):
```
Comment (takagi) [inline tests/cupy_tests/sparse_tests/test_csr.py]: Would you also check that `m` and `n` are not the same instances?

File: tests/cupy_tests/sparse_tests/test_csc.py [hunk 2]
```diff
@@ -55,6 +55,28 @@ def test_indptr(self):
...
+        n = cupy.sparse.csc_matrix(self.m.tocsr())
+        cupy.testing.assert_array_equal(n.data, self.m.data)
+        cupy.testing.assert_array_equal(n.indices, self.m.indices)
+        cupy.testing.assert_array_equal(n.indptr, self.m.indptr)
+        self.assertEqual(n.shape, self.m.shape)
+
+    def test_copy(self):
+        n = self.m.copy()
+        self.assertIsInstance(n, cupy.sparse.csc_matrix)
+        cupy.testing.assert_array_equal(n.data, self.m.data)
```
Comment (takagi) [inline tests/cupy_tests/sparse_tests/test_csc.py]: Would you also check that `n.data` and `self.m.data` are copied and not the same instances, as well as `indices` and `indptr`.

File: tests/cupy_tests/sparse_tests/test_csr.py [hunk 3]
```diff
@@ -58,6 +58,28 @@ def test_indptr(self):
...
+        n = cupy.sparse.csr_matrix(self.m.tocsc())
+        cupy.testing.assert_array_equal(n.data, self.m.data)
+        cupy.testing.assert_array_equal(n.indices, self.m.indices)
+        cupy.testing.assert_array_equal(n.indptr, self.m.indptr)
+        self.assertEqual(n.shape, self.m.shape)
+
+    def test_copy(self):
+        n = self.m.copy()
+        self.assertIsInstance(n, cupy.sparse.csr_matrix)
+        cupy.testing.assert_array_equal(n.data, self.m.data)
```
Comment (takagi) [inline tests/cupy_tests/sparse_tests/test_csr.py]: Would you also check that `n.data` and `self.m.data` are copied and not the same instances, as well as `indices` and `indptr`.

**Comment by unnonouno** [comment]
I fixed what you commented.

**Comment by takagi** [comment]
LGTM!

**Pull request merged**